### PR TITLE
Keras IO example - GPT2 Text Generation with KerasNLP

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
the code was already in backend agnostic keras. Just updated imports.

### CODE REVIEW

File: examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+the backend of your choice. We select the JAX backend below.
+"""
+
+"""shell
+pip install git+https://github.com/keras-team/keras-nlp.git -q
+"""
+
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"  # or "tensorflow" or "torch"
```
Comment (fchollet) [inline examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py:42]: You can remove these 2 lines if it works with any backend.

File: examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+# Load chinese poetry dataset.
+git clone https://github.com/chinese-poetry/chinese-poetry.git
+"""
+
+"""
+Load text from the json file. We only use《全唐诗》for demo purposes.
+"""
+
+import os
+import json
```
Comment (fchollet) [inline examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py:316]: Import imports at top

File: examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+poem_collection = []
+for file in os.listdir("chinese-poetry/全唐诗"):
+    if ".json" not in file or "poet" not in file:
+        continue
+    full_filename = "%s/%s" % ("chinese-poetry/全唐诗", file)
```
Comment (fchollet) [inline examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py:322]: Use f-strings for formatting

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py (hunk 1 above)
Comment (divyashreepathihalli) [reply examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py]: done

File: examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py (hunk 2 above)
Comment (divyashreepathihalli) [reply examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py]: done

File: examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py (hunk 3 above)
Comment (divyashreepathihalli) [reply examples/keras_io/generative/gpt2_text_generation_with_kerasnlp.py]: done!

**Review by fchollet — APPROVED** [review]
LGTM

**Pull request merged**